instrumentation/apache-log4j-2/src/main/java/com/nr/agent/instrumentation/log4j2/AgentUtil.java

```diff
@@ -29,6 +29,9 @@ public class AgentUtil {
     public static final String ERROR_CLASS = "error.class";
     public static final String ERROR_STACK = "error.stack";
     public static final String THREAD_NAME = "thread.name";
+    public static final String THREAD_ID = "thread.id";
```

Comment:
Too late for changing this, but I think this was a really good use case for an enum.
```
public enum LogAttr {
  MESSAGE("message"),
  ...
  ERROR_CLASS("error.class");

  private final String label;

  LogAttr(String label) {
    this.label = label;
  }

  public toString() {
    return label;
  }
}
```
This would allow us to use HashMap, which uses less CPU, and usually also uses less memory.
And with this implementation, it should not require any additional code for jsonification.

Reply:
Interesting, this might be worth revisiting at a later time. Performance has been pretty good with how it currently is but it would be cool to see if we can improve that further.